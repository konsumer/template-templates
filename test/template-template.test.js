import tt from '../index.js'
import { readFile } from 'fs/promises'
import * as url from 'url';

const __dirname = url.fileURLToPath(new URL('.', import.meta.url));

// load the file contents
const tstring = await readFile(`${__dirname}/demo.tpl`, 'utf8')

// these are some vars I'm going to pass to the template
const newsBad = {
  name: 'Mr. Anderson',
  company: 'MegaCorp',
  agent: 'Agent Smith',
  news: 'bad',
  reason: 'your cat died.'
}

const newsGood = {
  name: 'Mr. Anderson',
  company: 'MegaCorp',
  agent: 'Agent Smith',
  news: 'good',
  reason: 'you won the lottery.'
}

// [ 'name', 'company', 'agent', 'news', 'reason' ]
const params = Object.keys(newsBad)

describe('template-template', () => {
  it('should be able to compile a template', () => {
    const template = tt.compile(tstring, params)
    expect(template(newsBad)).toMatchSnapshot()
  })
  
  it('should be able to compile a template, in immediate mode', () => {
    expect(tt(tstring, newsBad)).toMatchSnapshot()
  })
  
  it('should handle logic of good news', () => {
    expect(tt(tstring, newsGood)).toMatchSnapshot()
  })

  it('should be able to pre-compile templates for export', () => {
    const template = tt.compile(tstring, params)
    expect(template.toString()).toMatchSnapshot()
  })

  it('should be cool with backticks', () => {
    const tstring2 = 'This is a markdown string with `code`.'
    expect(tt(tstring2)).toMatchSnapshot()
  })
})