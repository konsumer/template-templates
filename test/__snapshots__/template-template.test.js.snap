// Vitest Snapshot v1

exports[`template-template > should be able to compile a template 1`] = `
"Hi Mr. Anderson,

We regret having to break this bad news to you, but your cat died.

Signed Your Eternal Friends at MegaCorp,
Agent Smith"
`;

exports[`template-template > should be able to compile a template, in immediate mode 1`] = `
"Hi Mr. Anderson,

We regret having to break this bad news to you, but your cat died.

Signed Your Eternal Friends at MegaCorp,
Agent Smith"
`;

exports[`template-template > should be able to pre-compile templates for export 1`] = `
"function anonymous(_tpltpl_vars
) {
const {name,company,agent,news,reason} = _tpltpl_vars
return \`Hi \${name},

\${news === 'good'
  ? 'We are delighted to inform you that'
  : 'We regret having to break this bad news to you, but'
} \${reason}

Signed Your Eternal Friends at \${company},
\${agent}\`
}"
`;

exports[`template-template > should be cool with backticks 1`] = `"This is a markdown string with \`code\`."`;

exports[`template-template > should handle logic of good news 1`] = `
"Hi Mr. Anderson,

We are delighted to inform you that you won the lottery.

Signed Your Eternal Friends at MegaCorp,
Agent Smith"
`;
